File: apps/app-frontend/src/helpers/noctrinth-file-cache.ts
/**
 * The store of downloaded content files, kept by hash.
 *
 * See `packages/app-lib/src/util/noctrinth_file_cache.rs` for what goes in it:
 * a mod is downloaded once and copied out of the store for every pack, version
 * and instance that asks for it afterwards.
 */
import { invoke } from '@tauri-apps/api/core'

/** How many bytes of downloaded files are being kept. */
export async function fileCacheSize(): Promise<number> {
	return await invoke('plugin:noctrinth-file-cache|file_cache_size')
}

/** Empties the store. Everything in it can be downloaded again. */
export async function purgeFileCache(): Promise<void> {
	return await invoke('plugin:noctrinth-file-cache|file_cache_purge')
}

File: apps/app/src/api/noctrinth_file_cache.rs
//! The store of downloaded content files, kept by hash.
//!
//! See `theseus::file_cache` for what goes in it and why. These commands are
//! what the Resource management settings need to show its size and empty it.

use crate::api::Result;

pub fn init<R: tauri::Runtime>() -> tauri::plugin::TauriPlugin<R> {
    tauri::plugin::Builder::new("noctrinth-file-cache")
        .invoke_handler(tauri::generate_handler![
            file_cache_size,
            file_cache_purge,
        ])
        .build()
}

/// How many bytes of downloaded files are being kept.
#[tauri::command]
pub fn file_cache_size() -> u64 {
    theseus::file_cache::file_cache_size()
}

/// Empties the store. Everything in it can be downloaded again.
#[tauri::command]
pub async fn file_cache_purge() -> Result<()> {
    Ok(theseus::file_cache::purge_file_cache().await?)
}
